dev/NavigationView/NavigationView.cpp

```diff
@@ -1490,7 +1495,25 @@ void NavigationView::UpdatePaneLayout()
                         {
                             const auto footersActualHeight = footerItemsRepeater.ActualHeight();
                             const auto menuItemsActualHeight = menuItems.ActualHeight();
-                            if (totalAvailableHeight > menuItemsActualHeight + footersActualHeight)
+
+                            if (m_footerItemsSource.Count() == 0 && !IsSettingsVisible())
+                            {
+                                footerItemsScrollViewer.MaxHeight(0);
+                                if (const auto& separator = m_visualItemsSeparator.get())
```

Comment:
I'm wondering if there is a simpler logic to determine when to show the separator so we can avoid duplicating this code in each block. Could we do just one check at the end to determine visible/collapsed ?

Reply:
I don't really think there is a simpler way of determining this. We could introduce a function for setting the visibility, at least we would save on the if case. What do you think?